lib/Compiler.js

```diff
@@ -442,7 +451,18 @@ class Compiler {
 					});
 				});
 			});
-		});
+		};
+
+		if (this.idle) {
+			this.cache.endIdle(err => {
+				if (err) return finalCallback(err);
+
+				this.idle = false;
+				run();
+			});
+		} else {
+			run();
+		}
```

Comment:
@sokra sorry to put comments on an old PR. just wanted to point out something. this change altered the behavior of the compiler. and it's not mentioned as a breaking change in the docs.

Previously you could do this.

```javascript
const compiler = webpack(config);

// ... do other things, or pass the compiler to other functions

compiler.run(); // compiler starts here.
```

but now, compiler starts as soon as `webpack` is called. which is not a huge deal, but a significant change in behavior.

Similar change happened for `.watch` it would have started the compiler in watch mode before. now it doesn't work that way.